core/src/test/java/kafka/admin/ConfigCommandUnitTest.java

```diff
@@ -410,6 +448,432 @@ public void testOptionEntityTypeNames() {
+        List<String> defaultIpArgs = t.getKey();
+        Map<String, String> defaultIpEntry = t.getValue();
+        ClientQuotaEntity defaultIpEntity = new ClientQuotaEntity(defaultIpEntry);
+
+        List<String> deleteArgs = Arrays.asList("--delete-config", "connection_creation_rate");
+        Set<ClientQuotaAlteration.Op> deleteAlterationOps = new HashSet<>(Collections.singletonList(new ClientQuotaAlteration.Op("connection_creation_rate", null)));
+        Map<String, Double> propsToDelete = Collections.singletonMap("connection_creation_rate", 50.0);
+
+        List<String> addArgs = Arrays.asList("--add-config", "connection_creation_rate=100");
+        Set<ClientQuotaAlteration.Op> addAlterationOps = new HashSet<>(Collections.singletonList(new ClientQuotaAlteration.Op("connection_creation_rate", 100.0)));
+
+        verifyAlterQuotas(concat(singleIpArgs, deleteArgs), singleIpEntity, propsToDelete, deleteAlterationOps);
+        verifyAlterQuotas(concat(singleIpArgs, addArgs), singleIpEntity, Collections.emptyMap(), addAlterationOps);
+        verifyAlterQuotas(concat(defaultIpArgs, deleteArgs), defaultIpEntity, propsToDelete, deleteAlterationOps);
+        verifyAlterQuotas(concat(defaultIpArgs, addArgs), defaultIpEntity, Collections.emptyMap(), addAlterationOps);
+    }
+
+    @Test
+    public void shouldAddClientConfig() {
+        List<String> alterArgs = Arrays.asList("--add-config", "consumer_byte_rate=20000,producer_byte_rate=10000",
+            "--delete-config", "request_percentage");
+        Map<String, Double> propsToDelete = Collections.singletonMap("request_percentage", 50.0);
+
+        Set<ClientQuotaAlteration.Op> alterationOps = new HashSet<>(Arrays.asList(
+            new ClientQuotaAlteration.Op("consumer_byte_rate", 20000d),
+            new ClientQuotaAlteration.Op("producer_byte_rate", 10000d),
+            new ClientQuotaAlteration.Op("request_percentage", null)
+        ));
+
+        BiConsumer<Optional<String>, Optional<String>> verifyAlterUserClientQuotas = (userOpt, clientOpt) -> {
+            Map.Entry<List<String>, Map<String, String>> t = toValues(userOpt, ClientQuotaEntity.USER);
+            List<String> userArgs = t.getKey();
+            Map<String, String> userEntry = t.getValue();
+            t = toValues(clientOpt, ClientQuotaEntity.CLIENT_ID);
+            List<String> clientArgs = t.getKey();
+            Map<String, String> clientEntry = t.getValue();
+
+            List<String> commandArgs = concat(alterArgs, userArgs, clientArgs);
+            ClientQuotaEntity clientQuotaEntity = new ClientQuotaEntity(concat(userEntry, clientEntry));
+            try {
+                verifyAlterQuotas(commandArgs, clientQuotaEntity, propsToDelete, alterationOps);
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            }
+        };
+
+        verifyAlterUserClientQuotas.accept(Optional.of("test-user-1"), Optional.of("test-client-1"));
```

Comment:
Could we just pass strings to `verifyAlterQuotas`? This conversion is too hard to read.
```java
        verifyAlterQuotas(Arrays.asList("--entity-type", "users", "--entity-name", "test-user-1"),
                new ClientQuotaEntity(Collections.singletonMap("users", "test-client-1")), propsToDelete, alterationOps);
```

It seems to me above code is more readable ...

Reply:
Thanks for a review.

Reworked method to make it more readable:

- rename `toValues` -> `argsAndExpectedEntity`
- reduce number of intermediate variables.

Can you, please, take a look and tell me if we need to refactor this even more.

Reply:
CI looks OK for me. Please, take a look.